### ISSUE TITLE
[Paper Note] Rho-1: Not All Tokens Are What You Need, Zhenghao Lin+, arXiv'24, 2024.04

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#4925]
# URL
- https://arxiv.org/abs/2404.07965
# Authors
  - Zhenghao Lin
  - Zhibin Gou
  - Yeyun Gong
  - Xiao Liu
  - Yelong Shen
  - Ruochen Xu
  - Chen Lin
  - Yujiu Yang
  - Jian Jiao
  - Nan Duan
  - Weizhu Chen
# Abstract
  - Previous language model pre-training methods have uniformly applied a next-token prediction loss to all training tokens. Challenging this norm, we posit that "9l training". Our initial analysis examines token-level training dynamics of language model, revealing distinct loss patterns for different tokens. Leveraging these insights, we introduce a new language model called Rho-1. Unlike traditional LMs that learn to predict every next token in a corpus, Rho-1 employs Selective Language Modeling (SLM), which selectively trains on useful tokens that aligned with the desired distribution. This approach involves scoring pretraining tokens using a reference model, and then training the language model with a focused loss on tokens with higher scores. When continual pretraining on 15B OpenWebMath corpus, Rho-1 yields an absolute improvement in few-shot accuracy of up to 30% in 9 math tasks. After fine-tuning, Rho-1-1B and 7B achieved state-of-the-art results of 40.6% and 51.8% on MATH dataset, respectively - matching DeepSeekMath with only 3% of the pretraining tokens. Furthermore, when continual pretraining on 80B general tokens, Rho-1 achieves 6.8% average enhancement across 15 diverse tasks, increasing both efficiency and performance of the language model pre-training.
# Translation (by gpt-5-nano)
- 従来の言語モデルの事前学習手法は、すべての学習トークンに一様に次のトークン予測損失を適用してきた。この規範に挑戦し、我々は「9l training」を提唱する。初期の分析は言語モデルのトークンレベルの学習ダイナミクスを検討し、異なるトークンごとに異なる損失パターンが現れることを明らかにした。これらの洞察を活用して、Rho-1という新しい言語モデルを導入する。従来のLMがコーパス内のすべての次のトークンを予測することを学習するのとは異なり、Rho-1はSelective Language Modeling（SLM）を採用し、望ましい分布に沿って有用なトークンだけを選択的に学習する。このアプローチは、参照モデルを用いて事前学習トークンにスコアを付与し、スコアが高いトークンに対して焦点を当てた損失で言語モデルを訓練することを含む。15BのOpenWebMathコーパスで継続的な事前学習を行うと、Rho-1は9つの数学タスクにおいてfew-shot精度を最大で30%改善する絶対的な改善をもたらす。ファインチューニング後、Rho-1-1Bと7BはそれぞれMATHデータセットで40.6%と51.8%の最先端結果を達成し、DeepSeekMathに匹敵するのに前学習トークンのわずか3%しか使用していない。さらに、80Bの一般トークンで継続的な事前学習を行うと、Rho-1は15の多様なタスクで平均6.8%の改善を達成し、言語モデルの事前学習の効率と性能を高める。
# Summary (by gpt-4o-mini)
- 「9l training」を提唱し、トークンごとに異なる損失パターンを取り入れた新しい言語モデルRho-1を導入。Selective Language Modelingを採用し、望ましいトークンのみを選択的に学習。OpenWebMathコーパスでの事前学習を通じて、9つの数学タスクにおいてfew-shot精度を最大30%改善し、最先端のMATHデータセット結果を達成。80Bの一般トークンでも多様なタスクで平均6.8%の性能向上を実現。

**Comment by AkihikoWatanabe** [comment]
日本語解説: https://techblog.cccmkhd.co.jp/entry/2024/12/10/081653

**Comment by AkihikoWatanabe** [comment]
openreview: https://openreview.net/forum?id=0NMzBwqaAJ
